lib/api/resource.go

```diff
@@ -25,10 +25,12 @@ func init() {
 	helpers = make(map[TypeMetadata]resourceHelper)
 	helpersByType = make(map[reflect.Type]resourceHelper)
 
-	registerHelper := func(t interface{}, tl interface{}) {
-		tmd := reflect.ValueOf(t).Elem().FieldByName("TypeMetadata").Interface().(TypeMetadata)
+	registerHelper := func(t Resource, tl Resource) {
+		// [smc] how about using an interface here rather than reflection?
+		tmd := t.GetTypeMetadata()
 		rh := resourceHelper{
 			tmd,
+			// [smc] I'm not up to speed on reflection, but can't you just use t here?
```

Comment:
The interface contained a pointer to the base struct - but for simplicity I wanted it to just contain the object - hence the Elem()

I was getting into problems having interfaces with pointers and didn't want the headache of thinking about it.

Certainly something we can re-investigate later - but since this is internal machinery I don't think it's worth jimmying with it now.